### ISSUE TITLE
As a user I want to praise anyone on Twitter 

### ISSUE THREAD

**Issue by kristoferlund** [givepraise/praise#1075]
![image](https://github.com/givepraise/praise/assets/9698363/372a4338-24b5-4cd1-b92a-50b507c0b9d6)

## Main success scenario

1. Write Tweet in the form `@givethpraise @user for doing something great`
2. TwitterBot picks up tweet from listening to all tweets mentioning `@givethpraise`
3. TwitterBot verifies that tweet has the correct format
4. TwitterBot looks up community host name in local cache and fetches list from API if not found
5. Discord Bot posts praise to API
    1. Header `user-agent`to let the `API` know which bot request came from
    2. Header `host` to let the `API` know which community the request came from
    3. Header `x-api-key` gives the bot access to call the `API`
        1. This key is configured in `ENV` and is not community specific
6. `API` looks up `host` and verifies that community is in state `active`
7. `API` makes changes to DB
8. TwitterBot shares praise message to praise bot feed (`@givethpraise` in this example)
9. TwitterBot tweets a reply to the praise tweet confirming it has been registered
10. Twitter notifies user that the praise message has a reply

## Extensions

1. Praise has not correct format
    1. Reply with error message, instructions, link to docs
2. Community not set up correctly
    1. Reply with error message, instructions, link to docs

**Comment by cmaxwe11** [comment]
Just to confirm, only registered users should be able to give praise. Currently, is that permission handled in the community DB or the Discord bot?

**Comment by RamRamez** [comment]
@kristoferlund 

"5. Discord Bot posts praise to API"
I don't understand. do you mean Twitter bot?

"10. Twitter notifies user that the praise message has a reply"
This step will be done automatically by Twitter, am I right?

**Comment by RamRamez** [comment]
@kristoferlund In the Discord bot we have an account activation step that the user needs to activate and sign the message with their wallet before trying to praise.
Do we have this step for the Twitter bot? if so, is the process exactly like the Discord bot?

**Comment by kristoferlund** [comment]
@RamRamez 

5. Yes, my bad, Twitter bot of course.
10. Yes

The activation step will be slightly different for the Twitter bot. And it is not 100% decided yet how it will work. Item 3 in the usecase should be "Twitter bot verifies that user has activated their account and are allowed to praise"

My thought was that you could begin with the praise command without the activation step. Allow anyone to praise to begin with. And then we will add activation when I am back.

**Comment by kristoferlund** [comment]
@RamRamez, see suggested sourceId and sourceName definitions above. Ok?

**Comment by RamRamez** [comment]
@kristoferlund 
I have problems posting reply tweets for "Praise Given" and all types of errors. Twitter doesn't allow to post duplicate content.
The tweet ID is unique and I can include it in the tweet body, but I'm not sure if it's the best idea.

![image](https://github.com/givepraise/praise/assets/73441857/5e37e113-77b6-4cfd-a131-ba8c2bef9c88)

**Comment by kristoferlund** [comment]
> I have problems posting reply tweets for "Praise Given" and all types of errors. Twitter doesn't allow to post duplicate content.

It seems all replies need to contain some unique information. Take a look at https://threadreaderapp.com/. 

Mentioning the user and linking to praise dashboard perhaps is enough.

```
@user, Your praise have been sent! 🙏

Read it here: https://praise.tecommons.org/praise/64cabcea49fdcebef2b9b0ba
```

<img width="612" alt="image" src="https://github.com/givepraise/praise/assets/9698363/87ef2bbd-482c-462c-9001-99301e197cc1">

**Comment by RamRamez** [comment]
@kristoferlund Yes, that's a good idea. We can do this for successful praises. What about error messages?
